### PR TITLE
Add virtio-blk to KVM server

### PR DESCRIPTION
**Pull request by artek-koltun**
In this PR communication with a QEMU instance is automated to plug/unplug SPDK virtio-blk devices on CreateVirtioBlk/DeleteVirtioBlk

### CODE REVIEW

**Review by glimchb — APPROVED** [review]
lgtm. why waitForEvent only applied on DeleteVirtioBlkDevice ?

**Comment by artek-koltun** [comment]
>  lgtm. why waitForEvent only applied on DeleteVirtioBlkDevice ?

No event for virtio-blk creation is issued

**Comment by artek-koltun** [comment]
A small update. SPDK creates vhost sockets by default in pwd directory of SPDK process. It is better to keep our ctrlr_dir option empty to make users to specify it.

File: pkg/kvm/monitor.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+
+	return nil
+}
+
+func (m *monitor) DeleteVirtioBlkDevice(id string) error {
+	err := m.rmon.DeviceDel(id)
+	if err != nil {
+		return fmt.Errorf("couldn't delete device: %w", err)
+	}
+	return m.waitForEvent("DEVICE_DELETED", id)
```
Comment (glimchb) [inline pkg/kvm/monitor.go:104]: maybe event didn't come but device was deleted... something to consider in the future...

**Comment by artek-koltun** [comment]
> I'm fine with leaving it empty.
> but take a look at https://github.com/opiproject/opi-spdk-bridge/blob/main/docker-compose.yml#L29 we do specify it in compose

in compose yes (we can also add explicit ctrlr_dir option in compose if we are going to adapt it to kvm case with qemu). But if a user wants to run it manually... it can be hard to understand what went wrong if spdk was run without `-S`

**Comment by glimchb** [comment]
> in compose yes (we can also add explicit ctrlr_dir option in compose if we are going to adapt it to kvm case with qemu). But if a user wants to run it manually... it can be hard to understand what went wrong if spdk was run without `-S`

agree

**Pull request merged**